src/components/pill/pill.test.js

```diff
@@ -0,0 +1,185 @@
+import * as React from "react";
+import Pill from "./pill.component";
+import { pillPreview } from "../../../cypress/locators/pill/index";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { closeIconButton } from "../../../cypress/locators/index";
+
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData = "cypress_data";
+
+// eslint-disable-next-line react/prop-types
+const PillComponent = ({ ...props }) => {
+  return (
+    <>
+      <Pill onDelete="noop" {...props}>
+        A pill is needed
+      </Pill>
+    </>
+  );
+};
+
+context("Testing Pill component", () => {
+  describe("should render Pill component with props", () => {
+    it.each(specialCharacters)(
+      "should render Pill using %s as label",
+      (label) => {
+        CypressMountWithProviders(<Pill>{label}</Pill>);
+
+        pillPreview().should("have.text", label);
+      }
+    );
+
+    it.each([
+      ["warning", "2px solid rgb(242, 133, 51)"],
+      ["neutral", "2px solid rgb(51, 91, 112)"],
+      ["negative", "2px solid rgb(205, 56, 75)"],
+      ["positive", "2px solid rgb(0, 138, 33)"],
+    ])(
+      "should render Pill component with colorVariant set to %s",
+      (color, output) => {
+        CypressMountWithProviders(
+          <Pill pillRole="status" colorVariant={color}>
+            Pill
+          </Pill>
+        );
+
+        pillPreview()
+          .should("have.css", "border", output)
+          .should("have.css", "background-color", "rgba(0, 0, 0, 0)");
+      }
+    );
+
+    it("should render Pill component with data-element", () => {
+      CypressMountWithProviders(<PillComponent data-element={testData} />);
+      pillPreview().should("have.attr", "data-element", testData);
+    });
+
+    it("should render Pill component with data-role", () => {
+      CypressMountWithProviders(<PillComponent data-role={testData} />);
+      pillPreview().should("have.attr", "data-role", testData);
+    });
+
+    it.each([
+      ["warning", "rgb(242, 133, 51)"],
+      ["neutral", "rgb(51, 91, 112)"],
+      ["negative", "rgb(205, 56, 75)"],
+      ["positive", "rgb(0, 138, 33)"],
+    ])(
+      "should render Pill component with color fill to %s",
+      (color, output) => {
+        CypressMountWithProviders(
+          <Pill pillRole="status" colorVariant={color} fill>
+            Pill
+          </Pill>
+        );
+
+        pillPreview()
+          // eslint-disable-next-line prefer-template
+          .should("have.css", "border", "2px solid " + output)
+          .should("have.css", "background-color", output);
+      }
+    );
+
+    it("should call the delete action after the CloseIcon is clicked", () => {
```

Comment:
let's move the `event` testing into a separate `describe`. And than just use `callback` initialisation only once in `beforeEach`

Reply:
Done